src/Payum/Core/Bridge/Propel/Model/NotificationDetails.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Payum\Core\Bridge\Propel\Model;
+
+use Payum\Core\Bridge\Propel\Model\om\BaseNotificationDetails;
+
+class NotificationDetails extends BaseNotificationDetails implements \ArrayAccess, \IteratorAggregate
```

Comment:
I don think we should have custom models like NotificationDetails, PaymentDetails. Just a basic one like ArrayObject. User can extend it in their code and give it meaningful name